scripts/container_ops.sh

```diff
@@ -64,39 +71,73 @@ run_cmd() {
 ###################################################################
 
 # >>>>>>>  Initialization
+# Input vars
+REGISTRY_LOGIN=${1:-""}
+PUSH_TO_REGISTRY=${2:-""}
 
+# Static vars
+# TODO(improve logic for finding app.conf): Currently we have locked config location in infra/app.conf, we need to expand this feature
+# in order if someone wants to build something in folder_name/ and it has different registry from infra/app.conf.
 APP_CONFIG_FILE="${GITHUB_WORKSPACE}/infra/app.conf"
-if [[ -f "${APP_CONFIG_FILE}" ]]; then
-  echo "Application Config detected! Loading parameters..."
-  # shellcheck source=${GITHUB_WORKSPACE}/infra/app.conf
-  source "${APP_CONFIG_FILE}"
-fi
-
-if [[ -z "${CONTAINER_REPO}" ]]; then
-  echo "Docker root repository is not defined. Aborting docker operations."
-  exit 1
-fi
-
-CONTAINER_IMG_TAG="${CONTAINER_REPO}:${APP_VERSION}"
-
-# >>>>>>>  Start
-
-# as first, build docker image
-run_cmd build "${1}"
-
-# as second, try to login
-if [[ "${3}" == "true" ]]; then
-  run_cmd login
-fi
-# as third, publish the image if not specified otherwise
-# Parameter ${2} is `push_image` { true | false }
-if [[ "${2}" == "true" ]]; then
-  run_cmd push
-else
-  echo "(push = ${2}) Docker push disabled by user, image will not pushed to the Registry"
-fi
-
-# finally, cleanup
-run_cmd cleanup
-
-# >>>>>>>  End
+CONTAINER_DIR="infra/"
+
+
+main(){
+ 
+  # Check if app.conf is available if not exit the script 
+  if [[ -f "${APP_CONFIG_FILE}" ]]; then
+    echo "Application Config detected! Loading parameters..."
+    # shellcheck source=${GITHUB_WORKSPACE}/infra/app.conf
+    source "${APP_CONFIG_FILE}"
+  
+  else
+
+    echo ""
+    echo "Application config is not available in ${APP_CONFIG_FILE}."
+    echo ">>> STOPPING! <<<"
+    echo ">>> STOPPING! <<<"
+    echo ""
+    exit 1
+    
+  fi
+
+
+  # Check if CONTAINER_REPO is defined in ${APP_CONFIG_FILE}
+  if [[ -z "${CONTAINER_REPO}" ]]; then
+    echo "Docker root repository is not defined. Aborting docker operations."
+    exit 1
+  fi
+
+  if [[ "${REGISTRY_LOGIN}" == "true" ]]; then
+    run_cmd login
+  fi
+
+  if [[ ${CONTAINER_REPO} == *"{container_prefix}"* ]];then
+
+      DOCKERFILES=$(cd "${CONTAINER_DIR}" || exit ; find * -name '*.Dockerfile')
+
+      for FILE in ${DOCKERFILES[@]} ; do
+        CONTAINER_IMG_TAG=$(create_container_tag "${FILE}")
+        run_cmd build "${CONTAINER_IMG_TAG}" "${FILE}"
+        run_cmd push "${CONTAINER_IMG_TAG}"
+      done
+
+  else
+      CONTAINER_IMG_TAG="${CONTAINER_REPO}:${APP_VERSION}"
+      run_cmd build "${CONTAINER_IMG_TAG}"
+      run_cmd push "${CONTAINER_IMG_TAG}"
+  fi
```

Comment:
We should refactor this part to be make more readable.
For instance, steps can be:
1) Determine all docker files (doesn't matter if we have one or 10 files
2) Create list
3) run build/push foreach dockerfile in the list
4) cleanup

Reply:
Just to clarify, in every case if we have just Dockerfile or api.Dockerfile, service.Dockerfile, Dockerfile we will go through those steps ? And we can remove flag "{container_prefix}" from the app.conf ?

Reply:
First part - yes. But we will keep `container_prefix` as a feature_flag -> if exist parse multiple files, if not parse Dockerfile only.